File: 0020_valid_parentheses/0020_valid_parantheses.cs
public class Solution {
    public bool IsValid(string s) {
        var stack = new Stack<char>();
        Dictionary<char, char> map = new Dictionary<char, char>() {
            { '(', ')'},
            { '[', ']'},
            { '{', '}'}
        };
        
        foreach (var c in s)
        {
            if (map.ContainsKey(c)) {
                stack.Push(c);
            }
            else if (stack.Count == 0 || map[stack.Pop()] != c) {
                return false;
            }
        }
        
        return stack.Count == 0 ? true : false;
    }
}

File: 0020_valid_parentheses/0020_valid_parantheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        opcl = dict(('()', '[]', '{}'))
        stack = []
        
        for idx in s:
            if idx in '([{':
                stack.append(idx)
            elif len(stack) == 0 or idx != opcl[stack.pop()]:
                return False
                
        return len(stack) == 0